### DISCUSSION TITLE
Is it possible to for the component to be on a custom object?

### DISCUSSION THREAD

**Discussion by isabelironafter** [deejay-hub/timeline-lwc#166] [Q&A] [answered]
I have installed the component and dragged it onto a page of a custom object from a managed package.
My first trial was to plot a related custom object but am getting the below error.

<img width="1335" alt="Bildschirmfoto 2022-01-20 um 11 32 37" src="https://user-images.githubusercontent.com/65859626/150326447-58ab4a76-5da4-4ad4-b190-920f373c85d7.png">

<img width="1318" alt="Bildschirmfoto 2022-01-20 um 12 02 55" src="https://user-images.githubusercontent.com/65859626/150326743-8f150486-0e12-4d22-9751-3c06f861b398.png">

**Comment by deejay-hub** [comment, marked as answer]
Hi @isabelironafter. Navigating the setup can be challenging. We really need to offer a series of setup screens to help users...so apologies for making this more difficult than it should be.

In short, yes, custom objects, including those in managed packages should be supported. You have done a great job at the setup with only 1 error in the metadata type that causes the error. The relationship name comes from the **Child Relationship Name** field on the field on inov8__PMT_Project__c that looks up to the parent object inov8__PMT_Program__c. So if I navigate to the Project object and find the lookup field this is what I see.

![image](https://user-images.githubusercontent.com/7021883/150338224-3627c99a-3aef-4e1f-8460-e8c9b9a9ee06.png)

So the Relationship Name field in the Timeline_Configuration__mdt needs to be **inov8__Projects__r**. You need to add the package namespace as the prefix and __r as the suffix. If the custom object is not in a managed package you don't need the prefix and if it's a standard relationship you don't need the suffix.

![image](https://user-images.githubusercontent.com/7021883/150338435-098e6d6e-7a5b-4818-bde2-eacb41ba83e0.png)

Once I configure this I can see the related projects if I drag the Time Warp component onto a parent program. Just make sure the initial start date is populated and has a value within your date range.

![image](https://user-images.githubusercontent.com/7021883/150338785-2a00238e-ddf8-41be-9d99-eacdb2c64edc.png)

Let me know how you get on.

**Reply by isabelironafter** [reply]
@deejay-hub thank you for your super quick reply, itt worked! 

Your Configuration Guide & Knowledge Hub are very very helpful. Setup videos would be the icing on the cake!

**Comment by Reecej88** [comment]
Hi @deejay-hub i'm using the same managed package as @isabelironafter and having similar difficulties with the relationship. Would you be able to assist? Some additional info on my use case:

1. Object API = inov8__PMT_Project__c
2. The "PMT Projects" object has been renamed from to "Onboarding Projects" within Salesforce 'Rename Tabs and Labels'
3. My association is between the inov8__PMT_Project__c object and the standard Account object via a custom Account__c field on the project.

I have tried various combinations of package, object and field combinations for the Relationship Name but nothing seems to pull through the record into the timeline. The object itself isn't visible within the filters.

![image](https://github.com/user-attachments/assets/ec5dc5dc-a927-47e7-8e9c-34fcc76c6715)
![image](https://github.com/user-attachments/assets/1b6ac9c0-8541-4aa6-9c8e-f1d745fa82b8)

Thanks very much in advance

**Reply by deejay-hub** [reply]
The position data field corresponds to a formula field. Does it return a Date or Date/Time field for the records you expect to see? You can either add the formula to the page to confirm it has a value or change the data to use CreatedDate which would always have a value.

**Reply by Reecej88** [reply]
Thanks for the quick reply. I've tried using the CreateDate but still nothing i'm afraid. The object doesn't appear in the list of records to filter so i'm assuming there is an issue with the relationship.
![image](https://github.com/user-attachments/assets/3e2b8dc3-ae66-42a4-823f-a76d7b088ab7)

**Reply by deejay-hub** [reply]
Are you using PersonAccounts?

**Reply by deejay-hub** [reply]
Assuming you're not using PersonAccounts then the relationship name doesn't need the prefix inov8__ in your example since you create it. You only need the package name if the field was included when installed.

<img width="273" alt="image" src="https://github.com/user-attachments/assets/47c44eeb-d838-46ef-8ee5-a50468fe58fe" />

Also check the value for position date always has a valid date.

**Reply by Reecej88** [reply]
Can confirm not using PersonAccounts.
Thanks very much for the information. I've now removed the prefix inov8__ as advised and can confirm the records now appear on the timeline!

**Comment by mknadeerjd** [comment]
Can I use it for CPQ Quotes, 

![git_Timewarp](https://github.com/user-attachments/assets/b4f0303f-b07c-4e6e-abfd-2a3de43b811c)

It's not showing up in the opportunity,
![image](https://github.com/user-attachments/assets/b5f278e5-481c-42df-9190-f00fa9dd433a)

Am I missing anything?

**Reply by deejay-hub** [reply]
Hi @mknadeerjd. Yes most custom objects are supported. 

If you follow the steps in the original post discussion it shows the steps needed to get the relationship name. Looks like "SBQQ_Opportunity2__c" might be the field name on the child object "SBQQ_Quote__c". You need to find the field on the child object then look for the "Child Relationship Name" field value. If the field was created when you installed the managed package then the syntax is "SBQQ__ChildRelName__r". 

Step 1: Navigate to SBQQ_Quote__c object in object manager
Step 2: Find and drilldown on the field SBQQ_Opportunity2__c and note the value in the "Child Relationship Name" field
Step 3: Update the relationship name in the Time Warp metadata to refer to the correct relationship name. Since the field is in a managed package the syntax is "SBQQ__yourvalue__r"

You can also remove the value in the metadata type related to Fallback Tooltip field. This is only needed if the object doesn;t support the UI API. In this case leaving it blank will use the compact layout of the object which is easier to add and remove fields from.
